src/modules/expenses/expense.routes.ts

```diff
@@ -0,0 +1,84 @@
+import { createRoute } from "@hono/zod-openapi";
+import { z } from "zod";
+
+import { AuthRoles } from "@/common/enums";
+import jsonContentRequired from "@/common/helpers/json-content-required.helper";
+import { jsonContent } from "@/common/helpers/json-content.helper";
+import {
+  authMiddleware,
+  checkRoleGuard,
+  requireAuth,
+} from "@/common/middlewares/auth.middleware";
+import * as HTTPStatusCodes from "@/common/utils/http-status-codes.util";
+import {
+  insertExpenseSchema,
+  selectExpenseSchema,
+} from "@/db/schemas/expense.model";
+
+const tags = ["Expenses"];
+
+export const createExpenseRoute = createRoute({
+  tags,
+  method: "post",
+  path: "/expenses",
+  middleware: [
+    authMiddleware(),
+    requireAuth(),
+    checkRoleGuard(AuthRoles.ADMIN, AuthRoles.USER),
+  ] as const,
+  request: {
+    body: jsonContentRequired(
+      insertExpenseSchema
+        .omit({
+          id: true,
+          createdAt: true,
+          updatedAt: true,
+          creatorId: true,
+        })
+        .partial({
+          payerId: true,
```

Comment:
We need the payerId from the frontend, because it could be that I am adding an expense which might be paid by someone else and not necessarily me, correct me if I misunderstood your reasoning for making it partial.